src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -91,12 +91,14 @@ if(CLR_CMAKE_TARGET_MACCATALYST)
 endif()
 
 if(CLR_CMAKE_TARGET_TVOS)
+    # with -fembed-bitcode passing -headerpad_max_install_names is not allowed so remove it from the CMake flags
     string(REPLACE "-Wl,-headerpad_max_install_names" "" CMAKE_C_LINK_FLAGS ${CMAKE_C_LINK_FLAGS})
     string(REPLACE "-Wl,-headerpad_max_install_names" "" CMAKE_CXX_LINK_FLAGS ${CMAKE_CXX_LINK_FLAGS})
     string(REPLACE "-Wl,-headerpad_max_install_names" "" CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS ${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS})
     string(REPLACE "-Wl,-headerpad_max_install_names" "" CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS ${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS})
-    add_compile_options(-fembed-bitcode)
-    add_link_options(-fembed-bitcode)
+    add_compile_options(-Wl,-no-undefined)
+    add_compile_options(-fembed-bitcode -fno-gnu-inline-asm)
+    add_link_options(-fembed-bitcode -fno-gnu-inline-asm)
```

Comment:
I came to the conclusion that `-fno-gnu-inline-asm` is unnecessary but also harmless.

Maybe `add_compile_options(-Wl,-no-undefined)` should just be `add_link_options(-no-undefined)`?

Reply:
So I tried `add_link_options(-no-undefined)` but clang then complained about not recognizing the flag which led me to pass it via `-Wl,-no-undefined`. However it turns out that the Apple linker ignores that one too, which is why the libraries build failed because it has an option set to error on unused command line options.

Apparently the replacement for the option is `-undefined error`, but that in turn resulted in this error:
```
ld: -undefined and -bitcode_bundle (Xcode setting ENABLE_BITCODE=YES) cannot be used together
```

We don't set `-bitcode_bundle` so I think it's implicit by the `-fembed-bitcode` which leads me to believe this is actually pointless.

---

The simulator build failed because we _do_ use inline asm there (which is allowed), so I moved it to mono.proj since checking for simulator isn't really straightforward in CMake. It means we don't set the flag on the libraries build but the chances of it using inline asm are quite slim :)

While doing that I noticed that we were actually missing to pass a bunch of options for the tvOS case in mono.proj, we only had the iOS case wired up so I fixed that.